### PR TITLE
Support multiple versions of Fast DDS in CI

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

File: .github/workflows/reusable-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+          - ubuntu-22.04
+          - windows-2019
+          - windows-2022
+
+    steps:
+
+      - name: Sync repository
+        uses: eProsima/eProsima-CI/external/checkout@v0
+        with:
+          path: ${{ github.workspace }}/src
```
Comment (irenebm) [inline .github/workflows/reusable-workflow.yml:90]:
```suggestion
          path: src
```

File: .github/workflows/reusable-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+# ASAN
+
+  asan:
+    runs-on: ubuntu-22.04
+    steps:
+
+      - name: Sync repository
+        uses: eProsima/eProsima-CI/external/checkout@v0
+        with:
+          path: ${{ github.workspace }}/src
```
Comment (irenebm) [inline .github/workflows/reusable-workflow.yml:131]:
```suggestion
          path: src
```

File: .github/workflows/reusable-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+# TSAN
+
+  tsan:
+    runs-on: ubuntu-22.04
+    steps:
+
+      - name: Sync repository
+        uses: eProsima/eProsima-CI/external/checkout@v0
+        with:
+          path: ${{ github.workspace }}/src
```
Comment (irenebm) [inline .github/workflows/reusable-workflow.yml:168]:
```suggestion
          path: src
```

File: .github/workflows/reusable-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+# CLANG
+
+  clang:
+    runs-on: ubuntu-22.04
+    steps:
+
+      - name: Sync repository
+        uses: eProsima/eProsima-CI/external/checkout@v0
+        with:
+          path: ${{ github.workspace }}/src
```
Comment (irenebm) [inline .github/workflows/reusable-workflow.yml:205]:
```suggestion
          path: src
```

File: .github/workflows/reusable-workflow.yml [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+  coverage:
+    runs-on: ubuntu-22.04
+    environment:
+      name: codecov
+    steps:
+
+      - name: Sync repository
+        uses: eProsima/eProsima-CI/external/checkout@v0
+        with:
+          path: ${{ github.workspace }}/src
```
Comment (irenebm) [inline .github/workflows/reusable-workflow.yml:235]:
```suggestion
          path: src
```

File: .github/workflows/reusable-workflow.yml [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+# FLAKY
+
+  flaky:
+    runs-on: ubuntu-22.04
+    steps:
+
+      - name: Sync repository
+        uses: eProsima/eProsima-CI/external/checkout@v0
+        with:
+          path: ${{ github.workspace }}/src
```
Comment (irenebm) [inline .github/workflows/reusable-workflow.yml:265]:
```suggestion
          path: src
```

File: .github/workflows/reusable-workflow.yml [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+        with:
+          os: ubuntu-22.04
+          cmake_build_type: Release
+          custom_version_build: ${{ inputs.custom_version_build }}
+          dependencies_artifact_postfix: ${{ inputs.dependencies_artifact_postfix }}
+          secret_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Compile and run tests
+        id: compile_and_test
+        uses: eProsima/eProsima-CI/multiplatform/colcon_build_test@main
```
Comment (irenebm) [inline .github/workflows/reusable-workflow.yml:278]:
```suggestion
        uses: eProsima/eProsima-CI/multiplatform/colcon_build_test_flaky@main
```

File: .github/workflows/reusable-workflow.yml [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+        with:
+          packages_names: ${{ env.code_packages_names }}
+          workspace_dependencies: ${{ github.workspace }}/install
+          ctest_args: --label-regex "xfail"
```
Comment (irenebm) [inline .github/workflows/reusable-workflow.yml:282]:
```suggestion
```

**Review by irenebm — CHANGES_REQUESTED** [review]

File: .github/workflows/reusable-workflow.yml (hunk 1 above)
Comment (rsanchez15) [reply .github/workflows/reusable-workflow.yml:90]: Addressed in a074d49 (applied everywhere)

File: .github/workflows/reusable-workflow.yml (hunk 7 above)
Comment (rsanchez15) [reply .github/workflows/reusable-workflow.yml:278]: I'll remove colcon_build_test_flaky from eProsima-CI as soon as I update it on every repository as it is redundand and implies to maintain the same code twice

File: .github/workflows/reusable-workflow.yml (hunk 8 above)
Comment (rsanchez15) [reply .github/workflows/reusable-workflow.yml:282]: Same comment as before

**Review by irenebm — APPROVED** [review]
LGTM

**Pull request merged**